### ISSUE TITLE
Weights without dense-matrix solve?

### ISSUE THREAD

**Issue by stevengj** [RJDennis/SmolyakApprox.jl#5]
We are currently running into problems with the fact that the [`smolyak_weights` function](https://github.com/RJDennis/SmolyakApprox.jl/blob/75f260cc2269b5165622cb5dec953b8abf7c85d7/src/smolyak_weights.jl#L1-L65) performs a large dense solve.  For four dimensions and µ=10 levels, we have 10⁵ points and we run out of memory for this matrix.

Is there any hope of an alternative algorithm that doesn't form such a large matrix explicitly?

cc @rpestourie

**Comment by RJDennis** [comment]
I think that this should be possible because the basis functions are Chebyshev polynominals.  Is this something that you need urgently?

**Comment by stevengj** [comment]
Right now it is the limiting factor in our usage of this package. 

Yes, I was hoping that some FFT-based algorithm would be possible, but I don't have any experience with this in the context of sparse grids.  Do you know of any good reference or any library that implements this?

**Comment by RJDennis** [comment]
I have just pushed some code that I think will help you.  It's all compatible with version 0.7/1.0.  The key new functions that you will want to use are smolyak_grid_full, smolyak_weights_full, and smolyak_evaluate_full.  These functions work with all the grid points at all layers rather than just the unique grid points, so I'm sure that they can be made even more efficient.  I think that everything is working and I've tested the new functions alongside the existing routines.  As yet only the isotropic grid case is coded.  I haven't updated the readme yet so let me know if anything is unclear.  I'd love to hear what you are using the package for.
